src/pages/Debug/DebugDetails.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {useState} from 'react';
+import type {OnyxKey} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import Button from '@components/Button';
+import ScrollView from '@components/ScrollView';
+import Text from '@components/Text';
+import TextInput from '@components/TextInput';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+
+type DebugDetailsProps = {
+    data: Record<string, unknown>;
+    onyxKey: OnyxKey;
+    isCollection?: boolean;
+    idSelector?: (data: Record<string, unknown>) => string;
+};
```

Comment:
```suggestion
type DebugDetailsProps = {
    data: Record<string, unknown>;
};
```

After spent some time in this file trying to improve the typings, I think it's impossible to come up with a good solution the current implementation. I would rather decoupling a little bit here and allowing passing only the `data` prop. See my other comments.